server/src/main/java/org/apache/cassandra/sidecar/lifecycle/ProcessLifecycleProvider.java

```diff
@@ -18,38 +18,232 @@
+        for (Map.Entry<String, String> entry : params.entrySet())
+        {
+            if (entry.getKey().startsWith("cassandra."))
+            {
+                defaultJvmProperties.put(entry.getKey(), entry.getValue());
+            }
+            else if (entry.getKey().startsWith("env."))
+            {
+                defaultEnvVars.put(entry.getKey().replaceAll("env.", ""), entry.getValue());
+            }
+        }
+        this.lifecycleDir = params.get(OPT_STATE_DIR);
+        this.defaultCassandraHome = params.get(OPT_CASSANDRA_HOME);
+        if (lifecycleDir == null || lifecycleDir.isEmpty())
+        {
+            throw new ConfigurationException("Configuration property '" + OPT_STATE_DIR + "' must be set for ProcessLifecycleProvider");
+        }
+        if (defaultCassandraHome == null || defaultCassandraHome.isEmpty())
+        {
+            throw new ConfigurationException("Configuration property '" + OPT_CASSANDRA_HOME + "' must be set for ProcessLifecycleProvider");
+        }
     }
 
     @Override
     public void start(InstanceMetadata instance)
     {
-        throw new UnsupportedOperationException("Not implemented yet");
+        if (isCassandraProcessRunning(instance))
+        {
+            LOG.info("Cassandra instance {} is already running.", instance.host());
+            return;
+        }
+        startCassandra(instance);
     }
 
     @Override
     public void stop(InstanceMetadata instance)
     {
-        throw new UnsupportedOperationException("Not implemented yet");
-
+        if (!isCassandraProcessRunning(instance))
+        {
+            LOG.info("Cassandra instance {} is already stopped.", instance.host());
+            return;
+        }
+        stopCassandra(instance);
     }
 
     @Override
     public boolean isRunning(InstanceMetadata instance)
     {
-        throw new UnsupportedOperationException("Not implemented yet");
+        return isCassandraProcessRunning(instance);
+    }
+
+    private void startCassandra(InstanceMetadata instance)
+    {
+        ProcessRuntimeConfiguration runtimeConfig = getRuntimeConfiguration(instance);
+        try
+        {
+            String stdoutLocation = getStdoutLocation(runtimeConfig.instanceName());
+            String stderrLocation = getStderrLocation(runtimeConfig.instanceName());
+            String pidFileLocation = getPidFileLocation(runtimeConfig.instanceName());
+            ProcessBuilder processBuilder = runtimeConfig.buildStartCommand(pidFileLocation,
+                                                                            stdoutLocation,
+                                                                            stderrLocation);
+            LOG.info("Starting Cassandra instance {} with command: {}", runtimeConfig.instanceName(), processBuilder.command());
+
+            Process process = processBuilder.start();
+            process.waitFor(CASSANDRA_PROCESS_TIMEOUT_MS, TimeUnit.MILLISECONDS); // blocking call, make async?
```

Comment:
As you have called out, consider using some sort of a periodic check for process, Instead of blocking the thread. Likewise for the stop process below.

```suggestion
        vertx.setPeriodic(1000, timerId -> {
            if (isCassandraProcessRunning(instance)) {
                vertx.cancelTimer(timerId);
                LOG.info("Started Cassandra instance {} with PID {}", runtimeConfig.instanceName(), 
                         readPidFromFile(Path.of(pidFileLocation)));
            } else if (System.currentTimeMillis() > startTime + CASSANDRA_PROCESS_TIMEOUT_MS) {
                vertx.cancelTimer(timerId);
                LOG.error("Failed to start Cassandra instance {} within timeout", runtimeConfig.instanceName());
                // Handle timeout error
            }
        });
```

Reply:
My understanding is that `vertx.setPeriodic` will return immediatelly right ? In which case, `LifecycleProvider.start` could return before the Cassandra process is initialized.

The issue is that `LifecycleManager.submitStartTask` assumes `LifecycleProvider.start` is synchronous and will only return after the process is guaranteed to be started. Are you suggesting making `LifecycleProvider.start` asynchronous and return a `Future` instead? Since this is a departure from the current synchronous design and may require many changes I would prefer to do this in a follow-up ticket if that works for you.

Reply:
Yep, valid point about the architectural impact of going fully async. Follow-up ticket makes sense.

I was hoping to move away from blocking the worker thread  from waiting on a kernel system call, towards a Java synchronization primitive (like CountDownLatch), which is more efficient and interruptible.  Though I recognize this would be an intermediate step - if we're planning to eventually adopt an async interface. We'd need to weigh whether this intermediate improvement justifies the additional churn.

Additionally `process.waitFor(timeout)` returns `false` when it times out, but we don't check the return value. Wondering if it is worth detecting the status of the process start for better failure handling.